cedar-policy/src/api.rs

```diff
@@ -440,11 +441,12 @@ impl ResidualResponse {
 /// Errors encountered while evaluating policies or expressions, or making
 /// authorization decisions.
 #[derive(Debug, Clone, PartialEq, Eq, Error)]
-pub enum EvaluationError {
+pub enum ApiEvaluationError {
     /// Error message, as string.
     /// TODO in the future this can/should be the actual Core `EvaluationError`
+    /// [+] Modifi
     #[error("{0}")]
-    StringMessage(String),
+    CoreError(#[from] EvaluationError),
```

Comment:
This variant name is public.  We generally more-or-less try to hide the distinction between Core and this crate from this crate's consumers.  Could we name the variant `EvaluationError` or something instead?

That also raises the question, do we need this enum at all vs. just using Core's `EvaluationError` directly?